include/zephyr/senss/senss.h

```diff
@@ -0,0 +1,384 @@
+ * Support multiple Application clients for open same sensor instance,
+ * in this case, the returned handle will different for different clients.
+ *
+ * @param type The sensor type which need to open.
+ *
+ * @param sensor_index The index which need to open.
+ *
+ * @param *handle The opened instance handle, if failed will be set to -1.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_open_sensor(
+		int type, int sensor_index,
+		int *handle);
+
+/**
+ * @brief Close sensor instance.
+ *
+ * @param handle The sensor instance handle need to close.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_close_sensor(
+		int handle);
+
+
+/**
+ * @brief Get current report interval of given client handle.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param value The data buffer to receive value (micro seconds unit).
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_get_interval(
+		int handle,
+		uint32_t *value);
+
+/**
+ * @brief Set current report interval of given client handle.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param value The value to be set (micro seconds unit).
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int senss_set_interval(
+		int handle,
+		uint32_t value);
+
+/**
+ * @brief Set current data change sensitivity of given client handle.
+ *
+ * Since each sensor type may have multiple data fields for it's value, this
+ * API support set separated sensitivity for each data field, or global
+ * sensitivity for all data fields.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The value to be set.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int senss_set_sensitivity(
+		int handle,
+		int index, uint32_t value);
+
+/**
+ * @brief Get current data change sensitivity of given client handle.
+ *
+ * Since each sensor type may have multiple data fields for it's value, this
+ * API support get separated sensitivity for each data field, or global
+ * sensitivity for all data fields.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The data buffer to receive value.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int senss_get_sensitivity(
+		int handle,
+		int index, uint32_t *value);
+
+/**
+ * @brief Read a data sample
+ *
+ * This API will trigger target sensor read a sample each time.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param buf The data buffer to receive sensor data sample.
+ *
+ * @param size The buffer size in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_read_sample(
+		int handle,
+		void *buf, int size);
```

Comment:
We shouldn't have 2 different data pipelines, the "one-shot" should just call the `senss_data_event_t` function set in `senss_register_data_event_callback`

Reply:
Yes, `one-short` can go `senss_data_event_t` function and internally we only have single pipeline. 

`senss_read_sample` is a pulling API, can be used in some cases like to get initial data of a sensor before user open the sensor, for data streaming, `senss_data_event_t` function is the recommend method which is more efficient. 

`senss_read_sample` internally can start streaming and when get the first sample stop the streaming.

Reply:
This gets a little tricky and is why I've been wanting to wait off on the subsystem until the kernel level drivers are done. We're moving away from blocking calls (even for "one-shot" samples). In general, the configurations should support the various configurations set in [enum chreSensorConfigureMode](https://cs.android.com/android/platform/superproject/+/master:system/chre/chre_api/include/chre_api/chre/sensor.h;drc=6951984bbefb96423970b82005ae381065e36704;bpv=1;bpt=1;l=468?q=system%2Fchre%2Fchre_api%2Finclude%2Fchre_api%2Fchre%2Fsensor.h&ss=android%2Fplatform%2Fsuperproject&gsn=chreSensorConfigureMode&gs=kythe%3A%2F%2Fandroid.googlesource.com%2Fplatform%2Fsuperproject%3Flang%3Dc%252B%252B%3Fpath%3Dsystem%2Fchre%2Fchre_api%2Finclude%2Fchre_api%2Fchre%2Fsensor.h%23chreSensorConfigureMode%2523n%2523dW0p0JHFDV8).

What I'm thinking is, you'll never have a since client that wants both streaming and one-shot data. That doesn't really make sense, if they're streaming and need the data ASAP, they should just flush. So I guess I would actually prefer to remove this API and tell developers to just reconfigure the sensor to a one-shot sample which will use the same data pipeline as streaming.

Reply:
@ghu0510 please address this

Reply:
removed useless API